testutils/src/main/java/com/amplifyframework/testutils/SynchronousDataStore.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.testutils;
+
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.core.ResultListener;
+import com.amplifyframework.core.model.Model;
+import com.amplifyframework.datastore.DataStoreItemChange;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * A utility to facilitate synchronous calls to the Amplify DataStore category.
+ * This is not appropriate for production use, but is valuable in test code.
+ * Test code wants to perform a series of sequential verifications, with the assumption
+ * that a DataStore operation has completed with some kind of terminal result.
+ */
+public final class SynchronousDataStore {
+    @SuppressWarnings("checkstyle:all") private SynchronousDataStore() {}
+
+    /**
+     * Gets an instance of the SynchronousDataStore.
+     * @return Synchronous DataStore instance
+     */
+    @NonNull
+    public static SynchronousDataStore instance() {
+        return new SynchronousDataStore();
```

Comment:
Same as for SynchronousApi(), can we make this singleton?

Reply:
Will do